study.md

```diff
@@ -21,18 +21,52 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+Interrupting a function by return; - when the code get executed to the "return;" what will trigger it to go back to print out "B" ? Also I would think that the result will not get any numbers with B then when return; gets in.
+
+function counter() {
+  for (var count = 1; ; count++) {  // infinite loop
+    console.log(count + "A"); // until 5
+      if (count === 5) {
+        return;
+      }
+      console.log(count + "B");  // until 4
+    }
+  console.log(count + "C");  // never appears
+}
+
+counter();
+
+// Output:
+// 1A
+// 1B
+// 2A
+// 2B
+// 3A
+// 3B
+// 4A
+// 4B
+// 5A
+
 ```
 
 Are there any nagging questions relating to functions that you have?
 
 ```md
-<!-- your answer here -->
+When you get an example of named function expression - I don't understand how come it doesn't follow the rule of if (n <= 1) {return 1;} else {return n * factorial(n - 1);}
+This is the confusing part about JavaScript.
```

Comment:
I'm not sure that I understand this question.  Feel free to ask me this in person, since it might be easier to clarify things.